docs: Document challenge types and when they are generated (#1495)
kantord/LibreLingo: docs/courses/skill.md

- `Skill > Id`: The ID of the course. **NOTE:** This should be unchanged if you're translating or editing existing course. Only if you're creating a new course, this should have a unique [UUID v4](https://www.uuidgenerator.net/version4) string. Details for which you can find [here](creating-courses.md).
- `Skill > Thumbnails`: A list of filenames of the thumbnails to be used on the course page to give an idea of the skill. A list of available files can be found on [`apps/web/static/images/`](https://github.com/kantord/LibreLingo/tree/main/apps/web/static/images). The names should be used without extension and without `_tiny` or `_tinier` parts. e.g. `banana2_tinier.jpg` should be written as `banana2`.

**`New words`** has a list of new words that the lesson teaches.
- `Word`: The word in the target language, i.e. the language the user is learning.
- `Synonyms`: A list of synonyms of the above word. (optional)
- `Translation`: Translation of the word in the source language, i.e. the language the user already knows.

> - `Also accepted`: A list of alternative translations to the source language, which will *not* be taught, but will be accepted as correct translations of the `Word` field if supplied by the user. Currently has no effect, but may be used by future challenge types. (optional)

Reviewer: for words, does this really have no effect? I think in short input challenges, users are asked to type out the translation of a word, so I think it's already used 🤔 _Phrases_ don't have this, but if I remember correctly, _Words_ do

```suggestion
- `Also accepted`: A list of alternative translations to the source language, which will *not* be taught, but will be accepted as correct translations of the `Word` field if supplied by the user. (optional)
```

Author: Unless I'm mistaken, short input challenges only ever prompt the user with the source language, and ask the user to type an answer in the target language. Therefore, either the `Word` or any of its `Synonyms` would be correct, but the `Also accepted` field is never used, as the prompt is always the `Translation`.

- `<destination language>`: A list of terms in destination language as key and meaning in target audience's language.
- `<target audience's language>`: A list of terms in target audience's language as key and meaning in destination language.

> ## How skills generate challenges

Reviewer: ```suggestion
## How skills are displayed to the user
```

> As the course author, you only have to specify the words and phrases you want to teach, and LibreLingo will generate the challenges (individual questions). However, it is useful to understand what challenges will be generated.

Reviewer: ```suggestion
As the course author, you have to specify the words and phrases you want to teach. LibreLingo will know how to present skills as a series of individual questions or challenges. It is useful to understand what challenges will be generated.
```
